src/nvim/ex_cmds.lua

```diff
@@ -2678,18 +2678,18 @@ return {
     func='ex_ni',
   },
   {
-    command='tearoff',
-    flags=bit.bor(NEEDARG, EXTRA, TRLBAR, NOTRLCOM, CMDWIN),
-    addr_type=ADDR_LINES,
-    func='ex_tearoff',
-  },
-  {
     command='terminal',
     flags=bit.bor(BANG, FILES, CMDWIN),
     addr_type=ADDR_LINES,
     func='ex_terminal',
   },
   {
+    command='tearoff',
+    flags=bit.bor(NEEDARG, EXTRA, TRLBAR, NOTRLCOM, CMDWIN),
+    addr_type=ADDR_LINES,
+    func='ex_tearoff',
+  },
```

Comment:
@noahfrederick It is strict alphabetical order for the first lowercase command letter. (*One* letter, cases where a built-in command starts with something, but lowercase letter are also an exception.) Following letters are not strict because

1. When looking up a command there is a quick jump table for one first letter only. (And only for one first lowercase letter.)
2. List is processed in a way that command that comes first has shorter minimal abbreviation.

@splinterofchaos  It is not list of functions in eval.txt which is looked up using bisection (BTW, why it was not changed to a hash?). Searching is O(n): direct iteration over a list, with a quick jump table listing first letters. Worse, some commands like `noautocmd` (modifier commands) are listed separately, and their effective abbreviations are also defined separately directly in ex_docmd.c, but they are not excluded from the regular search. Even worse, if you type `tearon` command it will search for it starting from `t` and till the end of the commands list (not till the next letter), so you may just as well put new `tearon` command after `~` and it still will be found.

Due to 2 it is sometimes reordered for the purposes of giving some command better abbreviation. Also new commands must be appended in a way that will not spoil existing abbreviations.